src/cli/commands/install.js

```diff
@@ -660,7 +660,7 @@ export class Install {
     }
 
     // build lockfile location
-    const loc = path.join(this.config.cwd, constants.LOCKFILE_FILENAME);
+    const loc = path.join(this.config.worktreeFolder || this.config.cwd, constants.LOCKFILE_FILENAME);
```

Comment:
Hm, I think right now it's fine, but maybe that's something that could be improved later. I was thinking that maybe we could add another variable, `workspaceFolder`, that would contain the path to the top-most workspace, and use it instead of `cwd` about everywhere (the difference with `cwd` would be that if you're in a subdirectory, then the `workspaceFolder` would go up until the `package.json` file). This way, people wouldn't have the issue of adding dependencies to their subfolders instead of their project, if they're in the wrong directory.

But that would be a BC, so we should probably discuss this in an other PR/RFC